docs/content/getting-started/crd-acme/index.md

```diff
@@ -0,0 +1,73 @@
+# Traefik & CRD & Let's Encrypt
+
+Traefik with an IngressRoute Custom Resource Definition for Kubernetes, and TLS through Let's Encrypt.
+{: .subtitle }
+
+This document is intended to be a fully working example demonstrating how to set up Traefik in [Kubernetes](https://kubernetes.io), with the dynamic configuration coming from the [IngressRoute Custom Resource](../../providers/kubernetes-crd.md), and TLS setup with [Let's Encrypt](https://letsencrypt.org). However, for the sake of simplicity, we're using [Rancher's k3s](https://github.com/rancher/k3s) for the Kubernetes cluster setup.
+
+In the following, the Kubernetes resources defined in YAML configuration files can be applied to the setup in two different ways: the first, and usual way, is simply with the ```kubectl apply``` command. The second, which can be used for this tutorial, is to directly place the files in the directory used by the rancher docker image for such inputs: `/var/lib/rancher/k3s/server/manifests`.
+
+## k3s docker-compose configuration
+
+Our starting point is the docker-compose configuration file, to start the rancher k3s cluster. You can start it with:
+
+```
+docker-compose -f rancher-k3s.yml up
+```
+
+```yaml
+--8<-- "content/getting-started/crd-acme/rancher-k3s.yml"
+```
+
+## cluster resources
+
+Let's now have a look (in the order they should be applied, if using ```kubectl apply```) at all the required resources for the full setup.
+
+First, the definition of the IngressRoute and the Middleware kinds. Also note the rbac authorization resources; they'll be referenced through the serviceAccountName of the deployment, later on.
+
+```yaml
+--8<-- "content/getting-started/crd-acme/01-crd.yml"
+```
+
+Then, the services. One for Traefik itself, and one for the app it routes for, i.e. in this case our demo HTTP server: [whoami](https://github.com/containous/whoami).
+
+```yaml
+--8<-- "content/getting-started/crd-acme/02-services.yml"
+```
+
+Next, the deployments, i.e. the actual pods behind the services. Again, one pod for Traefik, and one for the whoami app.
+
+```yaml
+--8<-- "content/getting-started/crd-acme/03-deployments.yml"
+```
+
+Now, as an exception to what we said above, please note that you should not let the ingressRoute resources below be applied automatically to your cluster. The reason is, as soon as the acme provider of Traefik detects we have TLS routers, it will try to generate the certificates for the corresponding domains. And this will not work, because as it is, our Traefik pod is not reachable from the outside, which will make the ACME TLS challenge fail. Therefore, for the whole thing to work, we must delay applying the ingressRoute resources until we have port-forwarding set up properly, which is the next step.
```

Comment:
Maybe we should add a note somewhere here, that upfront, you should have set up the chosen hostname to go against the public ip of the machine you're executing this? Otherwise TLS challenge will fail, right?

Reply:
Done.